### ISSUE TITLE
Charm fails to cluster squid/edge

### ISSUE THREAD

**Issue by gboutry** [canonical/charm-microceph#115]
Adding units to a deployed microceph fails with the following errors:

```
unit-microceph-1: 12:38:45 INFO unit.microceph/1.juju-log peers:2: {'snap-channel': 'squid/edge', 'public_net': IPv4Network('10.206.54.0/24'), 'cluster_net': IPv4Network('10.206.54.0/24'), 'micro_ip': IPv4Address('10.206.54.41')}
unit-microceph-1: 12:38:05 ERROR unit.microceph/1.juju-log peers:2: Failed executing cmd: ['microceph', 'cluster', 'join', 'eyJzZWNyZXQiOiIwOTNhYzhlMjdiNjhkMjgyYzZmMjNlYzczMjA1ZTIxNTkxZjJlYTgwNzdmMDcyZmRhNDY3ZGIyOGNkZTM2Mjc5IiwiZmluZ2VycHJpbnQiOiJjNGY2ZjUyZTE3NTUyMDg5ZGM5YjIzZDIwYWEyYTYyMGFlMmZhNTlhZjY0YjRjOGZmZjI1ZWJjMTFjNmQ2NTk2Iiwiam9pbl9hZGRyZXNzZXMiOlsiMTAuMjA2LjU0LjQwOjc0NDMiXX0=', '--microceph-ip', '10.206.54.41'], error: Error: 1 join attempts were unsuccessful. Last error: Joining server certificate SAN does not contain join token name

unit-microceph-1: 12:38:05 WARNING unit.microceph/1.juju-log peers:2: Error: 1 join attempts were unsuccessful. Last error: Joining server certificate SAN does not contain join token name
```

```
unit-microceph-1: 12:31:10 INFO unit.microceph/1.juju-log ceph:4: _on_relation_changed event
unit-microceph-1: 12:31:11 ERROR unit.microceph/1.juju-log ceph:4: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

unit-microceph-1: 12:31:11 WARNING unit.microceph/1.juju-log ceph:4: Microceph not bootstrapped yet.
unit-microceph-1: 12:31:11 INFO unit.microceph/1.juju-log ceph:4: Not processing request as service is not yet ready
unit-microceph-1: 12:31:11 INFO unit.microceph/1.juju-log ceph:4: _on_relation_changed event
unit-microceph-1: 12:31:11 ERROR unit.microceph/1.juju-log ceph:4: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

unit-microceph-1: 12:31:11 WARNING unit.microceph/1.juju-log ceph:4: Microceph not bootstrapped yet.
unit-microceph-1: 12:31:11 INFO unit.microceph/1.juju-log ceph:4: Not processing request as service is not yet ready
unit-microceph-1: 12:31:11 INFO juju.worker.uniter.operation ran "ceph-relation-changed" hook (via hook dispatching script: dispatch)
unit-microceph-0: 12:31:56 INFO juju.worker.uniter.operation ran "update-status" hook (via hook dispatching script: dispatch)
unit-microceph-1: 12:30:54 INFO unit.microceph/1.juju-log ceph:3: _on_relation_changed event
unit-microceph-1: 12:30:54 ERROR unit.microceph/1.juju-log ceph:3: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

unit-microceph-1: 12:30:54 WARNING unit.microceph/1.juju-log ceph:3: Microceph not bootstrapped yet.
unit-microceph-1: 12:30:54 INFO unit.microceph/1.juju-log ceph:3: Not processing request as service is not yet ready
unit-microceph-1: 12:30:54 INFO unit.microceph/1.juju-log ceph:3: _on_relation_changed event
unit-microceph-1: 12:30:55 ERROR unit.microceph/1.juju-log ceph:3: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

unit-microceph-1: 12:30:55 WARNING unit.microceph/1.juju-log ceph:3: Microceph not bootstrapped yet.
unit-microceph-1: 12:30:55 INFO unit.microceph/1.juju-log ceph:3: Not processing request as service is not yet ready
unit-microceph-1: 12:30:55 INFO juju.worker.uniter.operation ran "ceph-relation-changed" hook (via hook dispatching script: dispatch)
unit-microceph-1: 12:31:02 INFO unit.microceph/1.juju-log ceph:4: _on_relation_changed event
unit-microceph-1: 12:31:02 ERROR unit.microceph/1.juju-log ceph:4: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

unit-microceph-1: 12:31:02 WARNING unit.microceph/1.juju-log ceph:4: Microceph not bootstrapped yet.
unit-microceph-1: 12:31:02 INFO unit.microceph/1.juju-log ceph:4: Not processing request as service is not yet ready
unit-microceph-1: 12:31:02 INFO unit.microceph/1.juju-log ceph:4: _on_relation_changed event
unit-microceph-1: 12:31:02 ERROR unit.microceph/1.juju-log ceph:4: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

unit-microceph-1: 12:31:02 WARNING unit.microceph/1.juju-log ceph:4: Microceph not bootstrapped yet.
unit-microceph-1: 12:31:02 INFO unit.microceph/1.juju-log ceph:4: Not processing request as service is not yet ready
unit-microceph-1: 12:31:02 INFO unit.microceph/1.juju-log ceph:4: _on_relation_changed event
unit-microceph-1: 12:31:03 ERROR unit.microceph/1.juju-log ceph:4: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

unit-microceph-1: 12:31:03 WARNING unit.microceph/1.juju-log ceph:4: Microceph not bootstrapped yet.
unit-microceph-1: 12:31:03 INFO unit.microceph/1.juju-log ceph:4: Not processing request as service is not yet ready
unit-microceph-1: 12:31:03 INFO juju.worker.uniter.operation ran "ceph-relation-joined" hook (via hook dispatching script: dispatch)
unit-microceph-1: 12:31:10 INFO unit.microceph/1.juju-log ceph:4: _on_relation_changed event
unit-microceph-1: 12:31:10 ERROR unit.microceph/1.juju-log ceph:4: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

unit-microceph-1: 12:31:10 WARNING unit.microceph/1.juju-log ceph:4: Microceph not bootstrapped yet.
unit-microceph-1: 12:31:10 INFO unit.microceph/1.juju-log ceph:4: Not processing request as service is not yet ready
unit-microceph-1: 12:31:10 INFO unit.microceph/1.juju-log ceph:4: _on_relation_changed event
unit-microceph-1: 12:31:10 ERROR unit.microceph/1.juju-log ceph:4: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized
```

**Comment by gboutry** [comment]
Tokens created:
```
sudo microceph cluster sql 'select * from core_token_records;'
+----+-------------+------------------------------------------------------------------+--------------------------------+
| id |    name     |                              secret                              |          expiry_date           |
+----+-------------+------------------------------------------------------------------+--------------------------------+
| 1  | microceph/1 | 093ac8e27b68d282c6f23ec73205e21591f2ea8077f072fda467db28cde36279 | 2024-09-25T15:27:16.323155895Z |
| 2  | microceph/2 | b71520d2ca9cc207e3bb92c43e165955ab0e1eb203299d68ce66bf5682d5aaa6 | 2024-09-25T15:38:16.75749333Z  |
+----+-------------+------------------------------------------------------------------+--------------------------------+
```

Join name:
```
openssl x509 -in server.crt -nout -text
...
            X509v3 Subject Alternative Name: 
                DNS:bm1.lxd, IP Address:127.0.0.1, IP Address:0:0:0:0:0:0:0:1
...
```

Since Microcluster LTS, token name and join name must match.

**Comment by UtkarshBhatthere** [comment]
Thank you for the bug @gboutry :D

**Comment by gboutry** [comment]
Fix: #118
